students/shawn_degraw/session03/strformat_lab.py

```diff
@@ -0,0 +1,56 @@
+from decimal import *
+
+#Task One
+print("Task One")
+print("File_{:03d} :  {:.2f}, {:.2e}, {:.3g}".format(2, 123.4567, 10000, 12345.67))
+
+#Task Two
+print("\nTask Two")
+value1=2
+value2=123.4567
+value3=10000
+value4=12345.67
+#?? trying to get significance of 3
+getcontext().prec = 3 
+#??not sure about 3 significant figures for value4
+print(f"File_{str(value1).zfill(3)} :  {round(value2,2)}, {Decimal(value3):.2e}, {Decimal(value4):.2e}")
+
+#Task Three
+#??No commas? 
+print("\nTask Three")
+def formatter(in_tuple):
+    form_string = f'the {str(len(in_tuple))} numbers are:  {("{:2}" * len(in_tuple))}'
+    return form_string.format(*in_tuple)
+
+print(formatter((2,3,5,7,9)))
+
+
+#Task Four
+print("\nTask Four")
+print("{3:02} {4:02} {2:02} {0:02} {1:02}".format(4, 30, 2017, 2, 27))
+
+#Task Five
+print("\nTask Five")
+tasklist = ['oranges', 1.3, 'lemons', 1.1]
+
+print(f"The weight of an {tasklist[0][0:-1]} is {tasklist[1]} and the weight of a {tasklist[2][0:-1]} is {tasklist[3]}")
```

Comment:
Minor: No need to put `0` for the starting index of a slice. That is, `tasklist[0][:-1]` is just fine. 0 is the default starting index when it's not specified.